maze_transformer/utils/token_utils.py

```diff
@@ -46,24 +46,40 @@ def get_tokens_up_to_path_start(
         return tokens[:path_start_idx]
 
 
-def decode_maze_tokens_to_coords(
-    tokens: list[str],
-    mazedata_cfg,  # TODO: cannot type this right now because importing MazeDatasetConfig causes a circular import
-    when_noncoord: typing.Literal["except", "skip", "include"] = "skip",
-) -> list[str | tuple[int, int]]:
-    """given a list of tokens, decode the coordinate-tokens to a list of coordinates, leaving other tokens as-is"""
-    output: list[str | tuple[int, int]] = list()
-    for idx, tk in enumerate(tokens):
-        if tk in mazedata_cfg.token_node_map:
-            output.append(mazedata_cfg.token_node_map[tk])
+def apply_mapping(
+    iter: Iterable[Any],
+    mapping: dict[Any, Any],
+    when_missing: Literal["except", "skip", "include"] = "skip",
+) -> list[Any]:
+    """Given a list and a mapping, apply the mapping to the list"""
+    output = list()
+    for item in iter:
+        if item in mapping:
+            output.append(mapping[item])
```

Comment:
nit: you can `continue` after this statement and remove the `else:` to reduce the nesting a bit